src/components/inspectors/TimerExpression.vue

```diff
@@ -0,0 +1,381 @@
+    </template>
+  </div>
+</template>
+
+<script>
+import Datepicker from 'vuejs-datepicker';
+import moment from 'moment';
+
+const periods = {
+  'day': 'D',
+  'week': 'W',
+  'month': 'M',
+  'year': 'Y',
+};
+
+export default {
+  components: {
+    Datepicker,
+  },
+  props: {
+    value: String,
+    hasEnds: {
+      type: Boolean,
+      default: true,
+    },
+    hasRepeat: {
+      type: Boolean,
+      default: true,
+    },
+    repeatLabel: {
+      type: String,
+      default: 'Repeat every',
+    },
+    weekLabel: {
+      type: String,
+      default: 'Repeat on',
+    },
+  },
+  data() {
+    const date = moment().set('hour', 0).set('minutes', 0);
+    const hours = [];
+    for (let i = 0; i < 48; i++) {
+      hours.push(date.format('HH:mm'));
+      date.add(30, 'minutes');
+    }
+    return {
+      today: moment().format('YYYY-MM-DD'),
+      hours,
+      weekdays: [
+        //  ISO week date weekday number, from 1 through 7,
+        //  beginning with Monday and ending with Sunday.
+        {
+          day: 7,
+          initial: 'S',
+          selected: false,
+        },
+        {
+          day: 1,
+          initial: 'M',
+          selected: false,
+        },
+        {
+          day: 2,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 3,
+          initial: 'W',
+          selected: false,
+        },
+        {
+          day: 4,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 5,
+          initial: 'F',
+          selected: false,
+        },
+        {
+          day: 6,
+          initial: 'S',
+          selected: false,
+        },
+      ],
+      startDate: new Date(),
+      startTime: '00:00',
+      repeat: '1',
+      periodicity: 'week',
+      ends: 'never',
+      endDate: new Date(),
+      times: '1',
+    };
+  },
+  computed: {
+    expression() {
+      this.startDate;
+      this.startTime;
+      this.repeat;
+      this.periodicity;
+      this.selectedWeekdays;
+      this.ends;
+      this.endDate;
+      this.times;
+      const expression = this.makeTimerConfig();
+      return expression;
+    },
+    selectedWeekdays() {
+      const selected = [];
+      this.weekdays.forEach(weekday => {
+        weekday.selected ? selected.push(weekday.day) : null;
+      });
+      return selected;
+    },
+    sameDay() {
+      return this.selectedWeekdays.length === 1 && this.weekdays[this.startDate.getDay()].selected;
+    },
+  },
+  watch: {
+    value(value) {
+      this.parseTimerConfig(value);
+    },
+  },
+  methods: {
+    currentDaySelected(day) {
+      return this.startDate.getDay() === (day.day % 7);
+    },
+    updateStartDate(date) {
+      this.startDate = date;
+      this.update();
+    },
+    updateEndDate(date) {
+      this.endDate = date;
+      this.update();
+    },
+    update() {
+      this.$emit('input', this.expression);
+    },
+    parseDateExpression(exp) {
```

Comment:
I have not used moment to validate the value, because it prints a warning in the console, when an invalid date is used:
![image](https://user-images.githubusercontent.com/8028650/52811189-e2b9ba00-306a-11e9-9683-b035c84dd7f8.png)

Reply:
@caleeli If you pass a 2nd argument to the moment constructor (with the format you expect the date to be in), the warning should go away. This is because, even if the date is reported invalid, it may very well be in a valid format that moment just happens not to understand.